weave/ops_primitives/date.py

```diff
@@ -35,21 +35,63 @@ def datetimetd_sub(lhs, rhs):
 
 
 @op(
-    name="datetime-add",
+    name="datetime-addms",
     input_type={
-        "lhs": types.UnionType(types.Timestamp(), types.TimeDelta()),
+        "lhs": types.UnionType(types.Timestamp(), types.Number()),
         "rhs": lambda input_types: types.optional(
             types.Timestamp()
-            if types.TimeDelta().assign_type(input_types["lhs"])
-            else types.TimeDelta()
+            if types.Number().assign_type(input_types["lhs"])
+            else types.Number()
         ),
     },
     output_type=types.optional(types.Timestamp()),
 )
 def datetime_add(lhs, rhs):
-    if rhs == None:
+    if rhs == None or rhs < 0:
```

Comment:
Dislike doing this but TTL needs some kind of check to not add in calculate duration if ttlDuration = -1. Not sure if there is an Op that can used to be used to filter this value beforehand